tests/src/nodes/nomos.rs

```diff
@@ -195,7 +196,19 @@ impl Node for NomosNode {
                         )
                     })
                     .collect::<Vec<_>>();
-                let mut nodes = vec![Self::spawn(configs.swap_remove(0)).await];
+
+                let overlay = TreeOverlay::new(configs[0].consensus.overlay_settings.clone());
+                let next_leader = overlay.next_leader();
+                let next_leader_idx = ids
+                    .iter()
+                    .position(|&id| NodeId::from(id) == next_leader)
+                    .unwrap();
+
+                // Spawn the next leader first, so the leader can receive votes from
+                // all nodes that will be subsequently spawned.
+                // If not, the leader will miss votes from nodes spawned before itself.
+                // This issue will be resolved by devising the block catch-up mechanism in the future
+                let mut nodes = vec![Self::spawn(configs.swap_remove(next_leader_idx)).await];
                 let listening_addr = nodes[0].get_listening_address().await;
```

Comment:
Shouldn't this be `nodes[next_leader_idx].get_listening_address()` if we want next leader to be available to other nodes before spawning (as initial_peer)?

Reply:
Since the `nodes` vector has only one element at line 211~212, we should use `nodes[0]`. But yeah, it's little not intuitive. Let me refactor this now.

Reply:
Okay, now I see, we append to the nodes array.